src/transformers/models/zoedepth/modeling_zoedepth.py

```diff
@@ -0,0 +1,1349 @@
+
+        conditional_log_binomial = self.conditional_log_binomial[bin_configurations_name]
+        x = conditional_log_binomial(last, bin_embedding)
+
+        # Now depth value is Sum px * cx , where cx are bin_centers from the last bin tensor
+        out = torch.sum(x * bin_centers, dim=1, keepdim=True)
+
+        return out, domain_logits
+
+
+class ZoeDepthMetricDepthEstimationHead(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+
+        n_bins = config.n_bins
+        bin_embedding_dim = config.bin_embedding_dim
+        min_depth = config.min_depth
+        max_depth = config.max_depth
+        n_attractors = config.num_attractors
+        attractor_alpha = config.attractor_alpha
+        attractor_gamma = config.attractor_gamma
+        attractor_kind = config.attractor_kind
+        min_temp = config.min_temp
+        max_temp = config.max_temp
+        bin_centers_type = config.bin_centers_type
+
+        self.min_depth = min_depth
+        self.max_depth = max_depth
+        self.bin_centers_type = bin_centers_type
+
+        # Bottleneck convolution
+        bottleneck_features = config.bottleneck_features
+        self.conv2 = nn.Conv2d(bottleneck_features, bottleneck_features, kernel_size=1, stride=1, padding=0)
+
+        # Regressor and attractor
+        if self.bin_centers_type == "normed":
+            SeedBinRegressorLayer = ZoeDepthSeedBinRegressor
+            Attractor = ZoeDepthAttractorLayer
+        elif self.bin_centers_type == "softplus":
+            SeedBinRegressorLayer = ZoeDepthSeedBinRegressorUnnormed
+            Attractor = ZoeDepthAttractorLayerUnnormed
+
+        self.seed_bin_regressor = SeedBinRegressorLayer(
+            bottleneck_features, n_bins=n_bins, min_depth=min_depth, max_depth=max_depth
+        )
+        self.seed_projector = ZoeDepthProjector(bottleneck_features, bin_embedding_dim)
+
+        self.projectors = nn.ModuleList(
+            [ZoeDepthProjector(config.fusion_hidden_size, bin_embedding_dim) for _ in range(4)]
+        )
+        self.attractors = nn.ModuleList(
+            [
+                Attractor(
+                    bin_embedding_dim,
+                    n_bins,
+                    n_attractors=n_attractors[i],
+                    min_depth=min_depth,
+                    max_depth=max_depth,
+                    alpha=attractor_alpha,
+                    gamma=attractor_gamma,
+                    kind=attractor_kind,
+                )
+                for i in range(4)
+            ]
+        )
+
+        last_in = N_MIDAS_OUT + 1  # +1 for relative depth
+
+        # use log binomial instead of softmax
+        self.conditional_log_binomial = ZoeDepthConditionalLogBinomialSoftmax(
+            last_in, bin_embedding_dim, n_classes=n_bins, min_temp=min_temp, max_temp=max_temp
+        )
+
+    def forward(self, out, rel_depth):
+        outconv_activation = out[0]
+        bottleneck = out[1]
+        feature_blocks = out[2:]
```

Comment:
Rather than pass the inputs as a tuple like this, it's better to hae `outconv_activation`, `bottleneck` and `feature_blocks` are input arguments to the forward method. Same goes for `ZoeDepthMultipleMetricDepthEstimationHeads`